test/e2e/update.go

```diff
@@ -17,10 +17,25 @@ import (
 
 	mgmtredhatopenshift20220904 "github.com/Azure/ARO-RP/pkg/client/services/redhatopenshift/mgmt/2022-09-04/redhatopenshift"
 	mgmtredhatopenshift20230701preview "github.com/Azure/ARO-RP/pkg/client/services/redhatopenshift/mgmt/2023-07-01-preview/redhatopenshift"
+	"github.com/Azure/ARO-RP/pkg/cluster"
 	"github.com/Azure/ARO-RP/pkg/util/stringutils"
 )
 
 var _ = Describe("Update clusters", func() {
+	It("must replace the ARO operator's CredentialsRequest if it has been deleted", func(ctx context.Context) {
+		By("deleting the CredentialsRequest")
+		err := clients.DynamicKubernetes.Resource(cluster.CredentialsRequestGroupVersionResource).Namespace("openshift-cloud-credential-operator").Delete(ctx, "openshift-azure-operator", metav1.DeleteOptions{})
```

Comment:
We shouldn't need the dynamic client here.  We can use the `controller-runtime` client. We don't need to deal with unstructured objects in our e2e, we can just reference them as long as they're recognized by the apiserver schema.